www/js/survey/enketo/answer.js

```diff
@@ -133,23 +133,30 @@ angular.module('emission.survey.enketo.answer', [
   /**
    * resolve timestamps label from the survey response
    * @param {XMLDocument} xmlDoc survey answer object
+   * @param {object} trip trip object
    * @returns {object} object with `start_ts` and `end_ts`
    */
-  function resolveTimestamps(xmlDoc) {
+  function resolveTimestamps(xmlDoc, trip) {
     // check for Date and Time fields
     const date = xmlDoc.getElementsByTagName('Date')?.[0]?.innerHTML;
     const start = xmlDoc.getElementsByTagName('Start_time')?.[0]?.innerHTML;
     const end = xmlDoc.getElementsByTagName('End_time')?.[0]?.innerHTML;
 
     if (!date || !start || !end) return null; // if any of the fields are missing, return null
 
-    const momentStart = moment(date + 'T' + start);
-    const momentEnd = moment(date + 'T' + end);
+    let additionStartTs = moment(date + 'T' + start).unix();
+    let additionEndTs = moment(date + 'T' + end).unix();
 
-    // return unix timestamps in milliseconds
+    // if the start and end timestamps are within the same minute, use the trip start and end timestamps
+    if (additionStartTs - (additionStartTs % 60) == trip.start_ts - (trip.start_ts % 60))
+      additionStartTs = trip.start_ts;
+    if (additionEndTs - (additionEndTs % 60) == trip.end_ts - (trip.end_ts % 60))
+      additionEndTs = trip.end_ts;
```

Comment:
And what happens if we don't? Please update comment.

```
- the survey is only precise to the minute, while trips are precise to milliseconds
- if the start/end timestamps resolved from the survey response are within the same minute as the trip start/end, we will use the exact trip start/end
- this is to avoid addition timestamps being slightly off because of truncation issues
```

Reply:
done a285285d6743087375b522366fabb3168a2fdfa5